consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -114,71 +162,48 @@ func (e *EventHandler) OnReceiveProposal(proposal *model.Proposal) error {
 	log.Debug().Msg("proposal forwarded from compliance engine")
 
 	// ignore stale proposals
-	if block.View < e.forks.FinalizedView() {
+	if block.View <= e.forks.FinalizedView() {
```

Comment:
### ⚠️ 

equality here opens up potential edge cases! Would strongly suggest to not change that. The reason is that forks errors if the parent is unknown _unless_ the parent is _below_ the pruning threshold. 

Consider the following fork:
```
A ← B  ⋮ ← C ← D ← E
       ⋮ ↖ X ← Y ← Z
       ⋮
       pruning threshold 
```
Lets assume we already have block C, D, E. Block C is finalized, so Forks threw everything away _below_ C (but still retains C). Now block X comes in and then block Y. If you drop X, then forks will miss the parent for Y and error. ☠️ 

```suggestion
	if block.View < e.forks.FinalizedView() {
```
This is a general pattern we have often in the code, we _keep_ the lowest level. It doesn't harm, because it is only one extra vertex in the graph. But thereby, you guarantee that forks knows everything that is equal or above the pruning threshold. 

Subtle comment: your code would potentially work, because we organize the blocks by view and not by height. Though, we know  it is safe to use `<` while there is a regorous argument to be made why using `<=` is always safe and we don't have that argument yet.

Reply:
Ok, I've change this because for us it doesn't make sense to process blocks at finalized view